src/sinks/util/retries.rs

```diff
@@ -123,7 +123,7 @@ where
 
                 if let Some(expected) = error.downcast_ref::<L::Error>() {
                     if self.logic.is_retriable_error(expected) {
-                        warn!(message = "Retrying after error.", error = ?expected);
+                        warn!(message = "Retrying after error.", error = %expected);
```

Comment:
I verified the changes works fine with master branch as well. The one thing I am leaving for you guys to decide is the warning message format.

`warn!(message = "Retrying after error.", error = ?expected);` will print :
`WARN sink{component_kind="sink" component_name=test_out component_type=http}:request{request_id=0}: vector::sinks::util::retries: Retrying after error. error=CallRequest { source: hyper::Error(Connect, ConnectError("tcp connect error", Os { code: 61, kind: ConnectionRefused, message: "Connection refused" })) }`

But, `warn!(message = "Retrying after error.", error = %expected);` will print :
`WARN sink{component_kind="sink" component_name=test_out component_type=http}:request{request_id=0}: vector::sinks::util::retries: Retrying after error. error=Failed to make HTTP(S) request: error trying to connect: tcp connect error: Connection refused (os error 61)`

Reply:
We definitely prefer the "display" (non-debug) error when possible, so the latter is preferred.